src/libethash/internal.c

```diff
@@ -353,21 +352,51 @@ ethash_cache *ethash_light_acquire_cache(ethash_light_t light)
 	return ret;
 }
 
-ethash_full_t ethash_full_new(ethash_params const* params,
-							  ethash_cache const* cache,
-							  ethash_callback_t callback)
+ethash_full_t ethash_full_new(char const *dirname,
+	const ethash_h256_t *seed_hash,
```

Comment:
liana is absolutely right. at an absolute minimum, you should maintain good style.